src/client/Microsoft.Identity.Client/WsTrust/MexDocument.cs

```diff
@@ -225,6 +228,26 @@ private void SetPolicyEndpointAddresses(XContainer mexDocument)
             }
         }
 
+        private IEnumerable<XElement> FindElements(XContainer mexDocument, XNamespace xNamespace, string element)
+        {
+            IEnumerable<XElement> bindingElements = mexDocument.Elements().First().Elements(xNamespace + element);
+
+            //Unable to find metadata information in root node. Attempting to search document.
+            if (bindingElements.Count() == 0)
+            {
+                bindingElements = mexDocument.Elements().DescendantsAndSelf().Elements(xNamespace + element);
```

Comment:
Another thought. When processing a 3rd party IdP metadata, we always come to this line (3 times in this case) and end up enumerating all descendants each time. Perhaps it's better in the constructor to find the `wsdl:definitions` element and then just pass that around (to the `ReadPolicies`, `ReadPolicyBindings`, etc.) instead of the whole mex document.